sys/src/9/ufs/ffs_vfsops.c

```diff
@@ -1633,7 +1633,7 @@ ffs_vgetf(MountPoint *mp, ino_t ino, int flags, vnode **vpp, int ffs_flags)
 			// lrand() returns int32_t, which isn't ideal but it's
 			// all we have right now
 			// TODO HARVEY We need a u64 random number generator
-			ip->i_gen = -INT32_MIN + rand();
+			ip->i_gen = (int)-INT32_MIN + rand();
```

Comment:
int64_t

Reply:
Yeah, you're right. Also, it shouldn't be 0 at all, and that seems to be the case in my testing, so I need to fix something in newfs.